setup.py

```diff
@@ -379,6 +379,7 @@ def setup_package():
                 "src/nrnpython",
                 "src/nrnmpi",
             ],
+            extra_compile_args=["-std=c++17"],
```

Comment:
I find the various setup.py and setup.py.in to be a bit of a rabbit warren. Hopefully this is enough and doesn't break anything.